node/pkg/near/watcher.go

```diff
@@ -0,0 +1,369 @@
+                                                // use the chunk_id...  hmmmmm
+						v = r.Get("receipt_id")
+						if !v.Exists() {
+							logger.Info("receipt_id");
+							return nil;
+						}
+
+						id, err := base58.Decode(v.String())
+						if err != nil {
+							return err
+						}
+
+						if e.debug {
+							logger.Error("emitter: " + hex.EncodeToString(a[:]) + " id: " + hex.EncodeToString(id))
+						}
+
+						var txHash = eth_common.BytesToHash(id) // 32 bytes = d3b136a6a182a40554b2fafbc8d12a7a22737c10c81e33b33d1dcb74c532708b
+
+						v = logs.Get("data")
+						if !v.Exists() {
+							logger.Info("data");
+							return nil;
+						}
+						pl, err := hex.DecodeString(v.String())
+						if err != nil {
+							return err
+						}
+
+						v = body.Get("result.header.timestamp")
+						if !v.Exists() {
+							logger.Info("result.header.timestamp");
+							return nil;
+						}
+						ts := uint64(v.Uint()) / 1000000000;
+						observation := &common.MessagePublication{
+							TxHash:           txHash,
+							Timestamp:        time.Unix(int64(ts), 0),
+							Nonce:            uint32(logs.Get("nonce").Uint()), // uint32
+							Sequence:         logs.Get("seq").Uint(),
+							EmitterChain:     vaa.ChainIDNear,
+							EmitterAddress:   a,
+							Payload:          pl,
+							ConsistencyLevel: 0,
+						}
+
+						nearMessagesConfirmed.Inc()
+
+						logger.Info("message observed",
+							zap.Uint64("ts", ts),
+							zap.Time("timestamp", observation.Timestamp),
+							zap.Uint32("nonce", observation.Nonce),
+							zap.Uint64("sequence", observation.Sequence),
+							zap.Stringer("emitter_chain", observation.EmitterChain),
+							zap.Stringer("emitter_address", observation.EmitterAddress),
+							zap.Binary("payload", observation.Payload),
+							zap.Uint8("consistency_level", observation.ConsistencyLevel),
+						)
+
+						e.msgChan <- observation
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (e *Watcher) Run(ctx context.Context) error {
+	// an odd thing to broadcast...
+	p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDNear, &gossipv1.Heartbeat_Network{
+		ContractAddress: e.wormholeContract,
+	})
+
+	logger := supervisor.Logger(ctx)
+	errC := make(chan error)
+
+	logger.Info("Near watcher connecting to RPC node ", zap.String("url", e.nearRPC))
+
+	go func() {
+		timer := time.NewTicker(time.Second * 1)
+		defer timer.Stop()
+
+		if e.next_round == 0 {
+			finalBody, err := getFinalBlock(e)
+			if err != nil {
+				logger.Error("StatusAfterBlock", zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDNear, 1)
+				errC <- err
+				return
+			}
+			e.next_round = gjson.ParseBytes(finalBody).Get("result.chunks.0.height_created").Uint()
+		}
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDNear {
+					panic("invalid chain ID")
+				}
+				panic("Unimplemented")
```

Comment:
![what](https://www.reactiongifs.us/wp-content/uploads/2017/08/what.gif)

Reply:
I will fix this in a later PR once we are in testnet